src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Pkcs12CertBag.cs

```diff
@@ -91,6 +87,8 @@ private static byte[] EncodeBagValue(Oid certificateType, ReadOnlyMemory<byte> e
         {
             if (certificateType == null)
                 throw new ArgumentNullException(nameof(certificateType));
+            if (certificateType.Value == null)
+                throw new CryptographicException(SR.Argument_InvalidOidValue);
```

Comment:
Since this API hasn't shipped, I think the ArgumentException that would now otherwise be thrown is fine (so the test should change, vs the product)

Reply:
I added it to match the behavior of `AsnWriter` when the field is declared as `Oid` backing type. Should it ever be changed for any reason then this is what would get thrown with `AsnWriter.WriteObjectIdentifier(new Oid(null, null) `. I believe it makes sense to keep it consistent.

Reply:
Ah, that's a fair point.